src/Native/System.Native/pal_process.cpp

```diff
@@ -389,12 +389,20 @@ extern "C" int32_t GetPriority(PriorityWhich which, int32_t who)
 {
     // GetPriority uses errno 0 to show succes to make sure we don't have a stale value
     errno = 0;
+#ifdef PRIORITY_REQUIRES_INT_WHO
+    return getpriority(which, who);
+#else
     return getpriority(which, static_cast<id_t>(who));
+#endif
 }
 
 extern "C" int32_t SetPriority(PriorityWhich which, int32_t who, int32_t nice)
 {
+#ifdef PRIORITY_REQUIRES_INT_WHO
```

Comment:
Thanks @nguerrera! I was just trying to figure out what's going wrong (by debugging with the old fashion print way using `MESSAGE(STATUS ..)` construct). :)
This is fixed by d9cf2d8.
Incidentally, do you still want me to add that assertion for `id_t` width vs. `int32_t` (using `sizeof` operator)?